src/Azure/Orleans.Reminders.AzureStorage/Storage/AzureBasedReminderTable.cs

```diff
@@ -27,15 +29,37 @@ public sealed class AzureBasedReminderTable : IReminderTable
             this.loggerFactory = loggerFactory;
             this.clusterOptions = clusterOptions.Value;
             this.storageOptions = storageOptions.Value;
+            this.remTableManager = new RemindersTableManager(
+                this.clusterOptions.ServiceId,
+                this.clusterOptions.ClusterId,
+                this.storageOptions,
+                this.loggerFactory);
         }
 
-        public async Task Init()
+        public async Task StartAsync(CancellationToken cancellationToken)
         {
-            this.remTableManager = await RemindersTableManager.GetManager(
-                this.clusterOptions.ServiceId,
-                this.clusterOptions.ClusterId,
-                this.loggerFactory,
-                options: this.storageOptions);
+            while (!cancellationToken.IsCancellationRequested)
+            {
```

Comment:
We could as well refactor this to make it a bit more predictable with the following:

try
{
   while (true)
   {
      cancellationToken.ThrowIfCancellationRequested();

      // do something here
     
      if (allGood) break;
   }

   _initializationTask.TrySetResult();
}
catch (OperationCanceledException ex)
{
   _initializationTask.TrySetCanceled(ex.CancellationToken);
}
catch (Exception ex)
{
  _initializationTask.TrySetException(ex);
}